src/components/forms/ProxyDeviceForm.tsx

```diff
@@ -0,0 +1,303 @@
+        className: "no-border-top override-with-form",
+        configuration: (
+          <div
+            className={classnames("device-name", {
+              "u-text--muted": isNoneDevice,
+            })}
+          >
+            <b>{item.key}</b>
+          </div>
+        ),
+        inherited: (
+          <div className="p-text--small u-text--muted u-no-margin--bottom">
+            From: {item.source}
+          </div>
+        ),
+        override: isNoneDevice ? (
+          <Button
+            appearance="base"
+            type="button"
+            title="Reattach volume"
+            onClick={() => {
+              ensureEditMode(formik);
+              removeDevice(noneDeviceId, formik);
+            }}
+            className="has-icon u-no-margin--bottom"
+          >
+            <Icon name="connected"></Icon>
+            <span>Reattach</span>
+          </Button>
+        ) : (
+          <Button
+            appearance="base"
+            type="button"
+            onClick={() => {
+              ensureEditMode(formik);
+              addNoneDevice(item.key, formik);
+            }}
+            className="has-icon u-no-margin--bottom"
+            dense
+          >
+            <Icon name="disconnect"></Icon>
+            <span>Detach</span>
+          </Button>
+        ),
+      }),
+    );
+
+    Object.keys(item.proxy).forEach((key) => {
+      if (key === "name" || key === "type") {
+        return null;
+      }
+
+      inheritedRows.push(
+        getInheritedDeviceRow({
+          label: deviceKeyToLabel(key),
+          inheritValue: item.proxy[key as keyof typeof item.proxy],
+          readOnly: false,
+          isDeactivated: isNoneDevice,
+        }),
+      );
+    });
+  });
+
+  const customRows: MainTableRow[] = [];
+  formik.values.devices.forEach((formDevice, index) => {
+    if (formDevice.type !== "proxy") {
+      return;
+    }
+    const device = formik.values.devices[index] as LxdProxyDevice;
+
+    customRows.push(
+      getConfigurationRowBase({
+        className: "no-border-top custom-device-name",
+        configuration: (
+          <RenameDeviceInput
+            name={device.name}
+            index={index}
+            setName={(name) => {
+              ensureEditMode(formik);
+              void formik.setFieldValue(`devices.${index}.name`, name);
+            }}
+          />
+        ),
+        inherited: "",
+        override: (
+          <Button
+            className="u-no-margin--top u-no-margin--bottom"
+            onClick={() => {
+              ensureEditMode(formik);
+              removeDevice(index, formik);
+            }}
+            type="button"
+            appearance="base"
+            hasIcon
+            dense
+            title="Detach Proxy"
+          >
+            <Icon name="disconnect" />
+            <span>Detach</span>
+          </Button>
+        ),
+      }),
+    );
+
+    Object.keys(device).forEach((key) => {
```

Comment:
I think we should be more opinionated and write out the individual fields here. For several reasons, as 1. we want a specific ordering, and the Object.keys might give use a different one after saving and loading again. 2. We only want to represent some fields, not all supported ones. 3. To get the translation to dropdowns or adding help text for some fields, it will be easier or more straight forward, if the fields are listed explicitly.